### DISCUSSION TITLE
Squash older commits away. How do I do it?

### DISCUSSION THREAD

**Discussion by NadChel** [community/community#69810] [Programming Help] [answered]
### Body

`credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest` is actually the latest commit. I want to keep it and squash the other commits into nonexistence. How do I do it? The way I did it below appears incorrect
```
squash ce69851 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}                                                                                         
squash 071f998 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}                                                                                         
squash 80ffd72 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}                                                                                         
pick 5cd3dbe credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest   
```
I ran `git rebase -i HEAD~2`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by spenserblack** [comment, marked as answer]
Rebase todo items are read from the top down, and the `squash` instruction squashes into the *previous* commit (the one above). It seems like this is what you want to do:
```git-rebase-todo
pick ce69851 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}                                                                                         
squash 071f998 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}                                                                                         
squash 80ffd72 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}                                                                                         
squash 5cd3dbe credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest
```

Git will pause and give you a combined commit message of all squashed commits with the chance to edit. So you can squash these 4 commits together and then pick the 4th message as the "real" commit message.

You should definitely make sure you have a backup of the old history so that you can easily restore if your rebase doesn't go as planned.

**Reply by NadChel** [reply]
Thank you!
